src/server/src/main/java/io/cassandrareaper/storage/IDistributedStorage.java

```diff
@@ -56,4 +66,36 @@
 
   void storeNodeMetrics(UUID runId, NodeMetrics nodeMetrics);
 
+  /**
+   * Gets the next free segment from the backend that is both within the parallel range and the local node ranges.
+   *
+   * @param runId id of the repair run
+   * @param parallelRange list of ranges that can run in parallel
+   * @param ranges list of ranges we're looking a segment for
+   * @return an optional repair segment to process
+   */
+  Optional<RepairSegment> getNextFreeSegmentForRanges(
+      UUID runId, Optional<RingRange> parallelRange, List<RingRange> ranges);
+
+  List<GenericMetric> getMetrics(
+      String clusterName,
+      Optional<String> host,
+      String metricDomain,
+      String metricType,
+      long since);
+
+  void storeMetric(GenericMetric metric);
+
+  void storeCompactions(String clusterName, String host, List<Compaction> activeCompactions)
+      throws JsonProcessingException;
+
+  List<Compaction> listCompactions(String clusterName, String host)
+      throws JsonProcessingException, IOException;
+
+  void storeStreams(String clusterName, String host, Set<CompositeData> activeStreams)
+      throws JsonProcessingException;
+
+  Set<CompositeData> listStreamingOperations(String clusterName, String host)
+      throws JsonProcessingException, IOException;
```

Comment:
Some questions here:
 - do we really want to explode metric operations to individually defined interface methods here? 
 - do we want to expose `JsonProcessingException` and `IOException` here? (it goes against the grain of existing methods defined here, and i suspect is programmatic errors, not real checked exceptions defined by these behaviours)

Reply:
I made the suggested changes 👍